lib/client.js

```diff
@@ -256,7 +264,10 @@ class Client extends Dispatcher {
     }
 
     try {
-      const request = new Request(opts, handler)
+      console.log('Executing request')
+      const request = this[kMaxRedirections] > 0
+        ? new Request(opts, new RedirectHandler(this, { ...opts, origin: this[kUrl] }, handler))
```

Comment:
I also have to inject `origin` based on the sanitized `url` in order to construct the `request.history` properly.

Reply:
I'm not sure I follow this.